src/main/java/duke/Parser.java

```
@@ -0,0 +1,134 @@
+                String.format("The description of %s cannot be empty.", userInput[0]));
+        }
+
+        return userInput[1];
+    }
+
+    /**
+     * Checks if user has input a number and returns the number as an int
+     *
+     * @param description String format of an index input by user
+     * @return int format of the index
+     * @throws DukeException If index is not a number
+     */
+    private static int parseIndex(String description) throws DukeException {
+        try {
+            return Integer.parseInt(description);
+        } catch (NumberFormatException e) {
+            throw new DukeException("Please input a number!");
+        }
+    }
+
+    /**
+     * Creates Task object with given taskName and taskDescription, marked as not done
+     *
+     * @param taskName        Name of the Task to be created
+     * @param taskDescription Description of the Task to be created
+     * @return Task created
+     */
+    private static Task createTask(String taskName, String taskDescription) {
+        return createTask(taskName, taskDescription, false);
+    }
+
+    /**
+     * Creates Task object with given taskName and taskDescription and isDone
+     *
+     * @param taskName        Name of the Task to be created
+     * @param taskDescription Description of the Task to be created
+     * @param isDone          boolean indicating if the task is marked as done
+     * @return Task created
+     */
+    private static Task createTask(String taskName, String taskDescription, boolean isDone) {
+        String[] temp = taskDescription.split(" /", 2);
+
+        switch (taskName) {
+        case "todo":
+            return new Todo(taskDescription, isDone);
+        case "event":
+            return new Event(temp[0], temp[1].split(" ", 2)[1], isDone);
+        case "deadline":
+            return new Deadline(temp[0], temp[1].split(" ", 2)[1], isDone);
+        default:
+            // should not reach this part
+            return null;
```

Try not to return null here, you could instead throw an Exception, reason being that for possible future extensions of Duke, it would be harder to debug, especially when faced with `NullPointerException`.